crates/edr_provider/src/data/gas.rs

```diff
@@ -144,3 +145,68 @@ fn min_difference(lower_bound: u64) -> u64 {
         200
     }
 }
+
+/// Compute miner rewards for percentiles.
+pub(super) fn compute_rewards(
+    block: &dyn SyncBlock<Error = BlockchainError>,
+    reward_percentiles: &[RewardPercentile],
+) -> Result<Vec<U256>, ProviderError> {
+    if block.transactions().is_empty() {
+        return Ok(reward_percentiles.iter().map(|_| U256::ZERO).collect());
+    }
+
+    let base_fee_per_gas = block.header().base_fee_per_gas.unwrap_or_default();
+
+    let gas_used_and_effective_reward = block
+        .transaction_receipts()?
+        .iter()
+        .enumerate()
+        .map(|(i, receipt)| {
+            let transaction = &block.transactions()[i];
+
+            let gas_used = receipt.gas_used;
+            // gas price pre EIP-1559 and max fee per gas post EIP-1559
+            let gas_price = transaction.gas_price();
+
+            let effective_reward =
+                if let Some(max_priority_fee_per_gas) = transaction.max_priority_fee_per_gas() {
+                    cmp::min(max_priority_fee_per_gas, gas_price - base_fee_per_gas)
+                } else {
+                    gas_price.saturating_sub(base_fee_per_gas)
+                };
+
+            (gas_used, effective_reward)
+        })
+        .sorted_by(|(_, reward_first), (_, reward_second)| reward_first.cmp(reward_second))
+        .collect::<Vec<(_, _)>>();
+
+    // Ethereum block gas limit is 30 million, so it's safe to cast to f64.
+    let gas_limit = block.header().gas_limit as f64;
```

Comment:
Why not use `try_into()` with the comment in the `expect("")` such that this at least panics if the assumption is somehow invalidated?

Reply:
Good idea, but `f64` doesn't implement `TryFrom<u64>` unfortunately.